cube/src/main/java/zone/cogni/asquare/cube/index/InternalIndexingServiceUtils.java

```diff
@@ -140,6 +145,71 @@ static List<String> getUrisFromQuery(@Nonnull IndexingServiceContext context,
     return paginatedQuery.convertSingleColumnUriToStringList(results);
   }
 
+  /**
+   * Indexable URIs are URIs which can be indexed.
+   * In case URIs are passed which might not need to be indexed this method can filter them out.
+   *
+   * @param context          service which needs an IndexMethod instance
+   * @param collectionFolder of collection (in index)
+   * @param uris             candidate uris which might or might not be indexed
+   * @return list of uris which will be indexed
+   */
+  static List<String> getIndexableUris(IndexingServiceContext context,
+                                       CollectionFolder collectionFolder,
+                                       List<String> uris) {
+    if (!shouldCheckForIndexableUris(collectionFolder))
+      return uris;
+
+
+    List<List<String>> uriSubLists = Lists.partition(uris, 20);
+    List<String> indexableUris =
+            uriSubLists.stream()
+                       .flatMap(uriSubList -> selectIndexableUris(context, collectionFolder, uriSubList).stream())
```

Comment:
split in  (! import of Collection is needed here)
```suggestion
                       .map(uriSubList -> selectIndexableUris(context, collectionFolder, uriSubList))
                       .flapMap(Collection::stream)
```